packages/components/pie-modal/src/index.ts

```diff
@@ -216,6 +223,34 @@ export class PieModal extends RtlMixin(LitElement) implements ModalProps {
         `;
     }
 
+    /**
+     * Render leadingAction button depending on prop availability.
+     *
+     * 1. If the prop `leadingAction` is not provided, the button is not rendered.
+     * 2. If the prop `leadingAction` is provided but any of the optional properties
+     * are not provided, they fall back to their default values.
+     *
+     * @private
+     */
+    private renderLeadingAction (): TemplateResult | typeof nothing {
+        const { text = 'Confirm', variant = 'primary', ariaLabel } = this.leadingAction;
```

Comment:
I think having this default value of `"Confirm"` means the button still renders if no text is passed in?

Reply:
Ah yes that's my bad. I'll remove in the next pr.